uxarray/grid/neighbors.py

```diff
@@ -3,11 +3,236 @@
+            Indicates whether to query nodes in a breadth-first manner
+        sort_results : bool, default=True
+            Indicates whether distances should be sorted
+
+        Returns
+        -------
+        d : ndarray of shape (xyz.shape[0], k), dtype=double
+            Distance array that keeps the distances of the k-nearest neighbors to the entries from xyz in each row
+        ind : ndarray of shape (xyz.shape[0], k), dtype=INT_DTYPE
+            Index array that keeps the indices of the k-nearest neighbors to the entries from xyz in each row
+        """
+
+        if k < 1 or k > self._n_elements:
+            raise AssertionError(
+                f"The value of k must be greater than 1 and less than the number of elements used to construct "
+                f"the tree ({self._n_elements}).")
+
+        xyz = _prepare_cart_xyz_for_query(xyz)
+
+        d, ind = self._current_tree().query(xyz, k, return_distance, dualtree,
+                                            breadth_first, sort_results)
+
+        ind = np.asarray(ind, dtype=INT_DTYPE)
+
+        if xyz.shape[0] == 1:
+            ind = ind.squeeze()
+
+        # perform query with distance
+        if return_distance:
+            # only one pair was queried
+            if xyz.shape[0] == 1:
+                d = d.squeeze()
+
+            return d, ind
+
+        return ind
+
+    def query_radius(self,
+                     xyz: Union[np.ndarray, list, tuple],
+                     r: Optional[int] = 1.0,
+                     return_distance: Optional[bool] = True,
+                     count_only: Optional[bool] = False,
+                     sort_results: Optional[bool] = False):
+        """Queries the tree for all neighbors within a radius ``r``.
+
+        Parameters
+        ----------
+        xyz : array_like
+           coordinate pairs in cartesian (x, y, z) to query
+        r: distance in degrees within which neighbors are returned
+            r can be a single value , or an array of values of shape x.shape[:-1] if different radii are desired for each point.
+        return_distance : bool, default=False
+            Indicates whether distances should be returned
+        count_only : bool, default=False
+            Indicates whether only counts should be returned
+        sort_results : bool, default=False
+            Indicates whether distances should be sorted
+
+        Returns
+        -------
+        d : ndarray of shape (xyz.shape[0], k), dtype=double
+            Distance array that keeps the distances of all neighbors within some radius to the entries from xyz in each row
+        ind : ndarray of shape (xyz.shape[0], k), dtype=INT_DTYPE
+            Index array that keeps the indices of all neighbors within some radius to the entries from xyz in each row
+        """
+
+        if r < 0.0:
+            raise AssertionError(
+                f"The value of r must be greater than or equal to zero.")
+
+        r = np.deg2rad(r)
```

Comment:
I'd be careful with this.

The reason we did the `deg2rad` conversion is because our `BallTree` used the haversine distance, which works in radians.

Here we are using the mikowski distance in Cartesian space, I'm not sure what the physical meaning of our "r" is. Can you look into it?

Reply:
Yeah, I think you are right, it does make sense that r should be in whatever measurement you have for your input grid.